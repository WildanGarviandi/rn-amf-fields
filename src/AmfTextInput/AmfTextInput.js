import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { StyleSheet, Text, TextInput, View } from 'react-native'

class AmfTextInput extends Component {
  constructor(props) {
    super(props)

    this.state = {
      valid: false,
      message: ''
    }
  }

  componentDidMount() {
    this.validate()
  }

  onBlur = () => {
    this.validate()
  }

  onFocus = () => {
    if (this.props.validation && this.props.validation.required && this.props.value === null) this.props.onChangeText('')
  }

  inputValid = () => {
    const { validation, label } = this.props
    if (!validation) return ({status: true})

    // check if pristine
    if (this.props.value === null || this.props.value === undefined) return ({status: true})

    // check whether input is required
    if (validation.required) {
      if (this.props.value === '') return ({status: false, message: label + ' tidak boleh kosong'})
    }

    // check regex
    if (validation.regex && this.props.value !== '') {
      if (!validation.regex.test(this.props.value)) return ({status: false, message: 'Format tidak sesuai'})
    }

    // check whether input needs exact length
    if (validation.exactLength) {
      if (this.props.value.length !== parseInt(validation.exactLength)) return ({status: false, message: label + ' harus ' + validation.exactLength + ' karakter'})
    }

    // check whether input needs minimum length
    if (validation.minLength) {
      if (this.props.value.length < parseInt(validation.minLength)) return ({status: false, message: label + ' minimal ' + validation.minLength + ' karakter'})
    }

    // check whether input needs maximum length
    if (validation.maxLength) {
      if (this.props.value.length > parseInt(validation.maxLength)) return ({status: false, message: label + ' maksimum ' + validation.minLength + ' karakter'})
    }

    return ({status: true})
  }

  validate = () => {
    let validationObject = this.inputValid()
    this.setState({valid: validationObject.status, errorMessage: validationObject.message})
  }

  render() {
    let textInputValidationStyle = {}
    let validationText
    
    if (!this.state.valid) {
      textInputValidationStyle = { borderColor: 'red' }
      validationText = <Text style={style.errorText}>{this.state.errorMessage}</Text>
    }

    return (
      <View>
        <Text style={style.label}>{this.props.label}</Text>
        <TextInput
          ref={(ref) => {this.textInput = ref}}
          {...this.props}
          onFocus={this.onFocus}
          onBlur={this.onBlur}
          underlineColorAndroid='transparent'
          style={[style.textInput, this.props.style, textInputValidationStyle]}
        />
        { validationText }
      </View>
    )
  }
}

const style = StyleSheet.create({
  errorText: {
    color: 'red',
    margin: 4,
    marginBottom: 0
  },
  label: {
    color: '#454545',
    marginBottom: 8
  },
  textInput: {
    backgroundColor: 'white',
    height: 40,
    padding: 10,
    borderWidth: 1,
    borderColor: '#DDDDDD',
    color: '#9B9B9B',
    fontSize: 14,
    borderRadius: 3,
    paddingBottom:10
  }
})

AmfTextInput.propTypes = {
  label: PropTypes.string.isRequired,
  onChangeText: PropTypes.func.isRequired,
  style: TextInput.propTypes.style,
  validation: PropTypes.object,
  value: PropTypes.string
}

export default AmfTextInput
