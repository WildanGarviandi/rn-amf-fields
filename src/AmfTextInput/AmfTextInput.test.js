import React from 'react'
import AmfTextInput from './AmfTextInput'
import renderer from 'react-test-renderer'

test('AmfInputText without validation', () => {
  let props = {
    label: 'Label',
    onChangeText: () => {},
    value: null
  }

  const component = renderer.create(<AmfTextInput {...props} />)

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('AmfInputText with required validation', () => {
  let props = {
    label: 'Label',
    onChangeText: () => {},
    value: null,
    validation: {
      required: true
    }
  }

  const component = renderer.create(<AmfTextInput {...props} />)
  const componentError = renderer.create(<AmfTextInput {...props} value="" />)
  const componentErrorAB = renderer.create(<AmfTextInput {...props} value="ab" />)

  let tree = component.toJSON()
  let treeError = componentError.toJSON()
  let treeErrorAB = componentErrorAB.toJSON()
  expect(tree).toMatchSnapshot()
  expect(treeError).toMatchSnapshot()
  expect(treeErrorAB).toMatchSnapshot()
})

test('AmfInputText with regex validation', () => {
  let props = {
    label: 'Label',
    onChangeText: () => {},
    value: null,
    validation: {
      regex: /e/
    }
  }
  let propsNotNull = {
    label: 'Label',
    onChangeText: () => {},
    value: '',
    validation: {
      regex: /e/
    }
  }
  let propsRight = {
    label: 'Label',
    onChangeText: () => {},
    value: 'test',
    validation: {
      regex: /e/
    }
  }
  let propsWrong = {
    label: 'Label',
    onChangeText: () => {},
    value: 'ups',
    validation: {
      regex: /e/
    }
  }

  const component = renderer.create(<AmfTextInput {...props} />)
  const componentNotNull = renderer.create(<AmfTextInput {...propsNotNull} />)
  const componentRight = renderer.create(<AmfTextInput {...propsRight} />)
  const componentWrong = renderer.create(<AmfTextInput {...propsWrong} />)

  let tree = component.toJSON()
  let treeNotNull = componentNotNull.toJSON()
  let treeRight = componentRight.toJSON()
  let treeWrong = componentWrong.toJSON()
  expect(tree).toMatchSnapshot()
  expect(treeNotNull).toMatchSnapshot()
  expect(treeRight).toMatchSnapshot()
  expect(treeWrong).toMatchSnapshot()
})

test('AmfInputText with minLength validation', () => {
  let props = {
    label: 'Label',
    onChangeText: () => {},
    value: null,
    validation: {
      minLength: 2
    }
  }

  const component = renderer.create(<AmfTextInput {...props} />)
  const componentError = renderer.create(<AmfTextInput {...props} value="" />)
  const componentErrorAB = renderer.create(<AmfTextInput {...props} value="ab" />)

  let tree = component.toJSON()
  let treeError = componentError.toJSON()
  let treeErrorAB = componentErrorAB.toJSON()
  expect(tree).toMatchSnapshot()
  expect(treeError).toMatchSnapshot()
  expect(treeErrorAB).toMatchSnapshot()
})

test('AmfInputText with exactLength validation', () => {
  let props = {
    label: 'Label',
    onChangeText: () => {},
    value: null,
    validation: {
      exactLength: 2
    }
  }

  const component = renderer.create(<AmfTextInput {...props} />)
  const componentError = renderer.create(<AmfTextInput {...props} value="" />)
  const componentErrorAB = renderer.create(<AmfTextInput {...props} value="ab" />)
  const componentErrorABC = renderer.create(<AmfTextInput {...props} value="abC" />)

  let tree = component.toJSON()
  let treeError = componentError.toJSON()
  let treeErrorAB = componentErrorAB.toJSON()
  let treeErrorABC = componentErrorABC.toJSON()
  expect(tree).toMatchSnapshot()
  expect(treeError).toMatchSnapshot()
  expect(treeErrorAB).toMatchSnapshot()
  expect(treeErrorABC).toMatchSnapshot()
})

test('AmfInputText with maxLength validation', () => {
  let props = {
    label: 'Label',
    onChangeText: () => {},
    value: null,
    validation: {
      maxLength: 2
    }
  }

  const component = renderer.create(<AmfTextInput {...props} />)
  const componentError = renderer.create(<AmfTextInput {...props} value="" />)
  const componentErrorAB = renderer.create(<AmfTextInput {...props} value="ab" />)
  const componentErrorABC = renderer.create(<AmfTextInput {...props} value="abC" />)

  let tree = component.toJSON()
  let treeError = componentError.toJSON()
  let treeErrorAB = componentErrorAB.toJSON()
  let treeErrorABC = componentErrorABC.toJSON()
  expect(tree).toMatchSnapshot()
  expect(treeError).toMatchSnapshot()
  expect(treeErrorAB).toMatchSnapshot()
  expect(treeErrorABC).toMatchSnapshot()
})

test('AmfInputText onFocus', () => {
  let props = {
    label: 'Label',
    onChangeText: jest.fn(),
    value: 'test',
    validation: {
      required: true
    }
  }
  let propsPristine = {
    label: 'Label',
    onChangeText: jest.fn(),
    value: null,
    validation: {
      required: true
    }
  }
  let propsNull = {
    label: 'Label',
    onChangeText: jest.fn(),
    value: null
  }

  // const component = renderer.create(<AmfTextInput {...props} />)
  const component = new AmfTextInput(props)
  const componentPristine = new AmfTextInput(propsPristine)
  const componentWithoutValidation = new AmfTextInput(propsNull)

  component.onFocus()
  componentPristine.onFocus()
  componentWithoutValidation.onFocus()

  expect(props.onChangeText).not.toHaveBeenCalled()
  expect(propsPristine.onChangeText).toHaveBeenCalled()
  expect(propsNull.onChangeText).not.toHaveBeenCalled()
})

test('AmfInputText onBlur', () => {
  const component = new AmfTextInput({})

  component.validate = jest.fn()
  component.onBlur()

  expect(component.validate).toHaveBeenCalled()
})
