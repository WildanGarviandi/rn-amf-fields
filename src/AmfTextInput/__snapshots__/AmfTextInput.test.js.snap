// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmfInputText with exactLength validation 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#454545",
        "marginBottom": 8,
      }
    }
  >
    Label
  </Text>
  <TextInput
    label="Label"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "#DDDDDD",
          "borderRadius": 3,
          "borderWidth": 1,
          "color": "#9B9B9B",
          "fontSize": 14,
          "height": 40,
          "padding": 10,
          "paddingBottom": 10,
        },
        undefined,
        Object {},
      ]
    }
    underlineColorAndroid="transparent"
    validation={
      Object {
        "exactLength": 2,
      }
    }
    value={null}
  />
</View>
`;

exports[`AmfInputText with exactLength validation 2`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#454545",
        "marginBottom": 8,
      }
    }
  >
    Label
  </Text>
  <TextInput
    label="Label"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "#DDDDDD",
          "borderRadius": 3,
          "borderWidth": 1,
          "color": "#9B9B9B",
          "fontSize": 14,
          "height": 40,
          "padding": 10,
          "paddingBottom": 10,
        },
        undefined,
        Object {
          "borderColor": "red",
        },
      ]
    }
    underlineColorAndroid="transparent"
    validation={
      Object {
        "exactLength": 2,
      }
    }
    value=""
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "red",
        "margin": 4,
        "marginBottom": 0,
      }
    }
  >
    Label harus 2 karakter
  </Text>
</View>
`;

exports[`AmfInputText with exactLength validation 3`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#454545",
        "marginBottom": 8,
      }
    }
  >
    Label
  </Text>
  <TextInput
    label="Label"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "#DDDDDD",
          "borderRadius": 3,
          "borderWidth": 1,
          "color": "#9B9B9B",
          "fontSize": 14,
          "height": 40,
          "padding": 10,
          "paddingBottom": 10,
        },
        undefined,
        Object {},
      ]
    }
    underlineColorAndroid="transparent"
    validation={
      Object {
        "exactLength": 2,
      }
    }
    value="ab"
  />
</View>
`;

exports[`AmfInputText with exactLength validation 4`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#454545",
        "marginBottom": 8,
      }
    }
  >
    Label
  </Text>
  <TextInput
    label="Label"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "#DDDDDD",
          "borderRadius": 3,
          "borderWidth": 1,
          "color": "#9B9B9B",
          "fontSize": 14,
          "height": 40,
          "padding": 10,
          "paddingBottom": 10,
        },
        undefined,
        Object {
          "borderColor": "red",
        },
      ]
    }
    underlineColorAndroid="transparent"
    validation={
      Object {
        "exactLength": 2,
      }
    }
    value="abC"
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "red",
        "margin": 4,
        "marginBottom": 0,
      }
    }
  >
    Label harus 2 karakter
  </Text>
</View>
`;

exports[`AmfInputText with maxLength validation 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#454545",
        "marginBottom": 8,
      }
    }
  >
    Label
  </Text>
  <TextInput
    label="Label"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "#DDDDDD",
          "borderRadius": 3,
          "borderWidth": 1,
          "color": "#9B9B9B",
          "fontSize": 14,
          "height": 40,
          "padding": 10,
          "paddingBottom": 10,
        },
        undefined,
        Object {},
      ]
    }
    underlineColorAndroid="transparent"
    validation={
      Object {
        "maxLength": 2,
      }
    }
    value={null}
  />
</View>
`;

exports[`AmfInputText with maxLength validation 2`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#454545",
        "marginBottom": 8,
      }
    }
  >
    Label
  </Text>
  <TextInput
    label="Label"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "#DDDDDD",
          "borderRadius": 3,
          "borderWidth": 1,
          "color": "#9B9B9B",
          "fontSize": 14,
          "height": 40,
          "padding": 10,
          "paddingBottom": 10,
        },
        undefined,
        Object {},
      ]
    }
    underlineColorAndroid="transparent"
    validation={
      Object {
        "maxLength": 2,
      }
    }
    value=""
  />
</View>
`;

exports[`AmfInputText with maxLength validation 3`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#454545",
        "marginBottom": 8,
      }
    }
  >
    Label
  </Text>
  <TextInput
    label="Label"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "#DDDDDD",
          "borderRadius": 3,
          "borderWidth": 1,
          "color": "#9B9B9B",
          "fontSize": 14,
          "height": 40,
          "padding": 10,
          "paddingBottom": 10,
        },
        undefined,
        Object {},
      ]
    }
    underlineColorAndroid="transparent"
    validation={
      Object {
        "maxLength": 2,
      }
    }
    value="ab"
  />
</View>
`;

exports[`AmfInputText with maxLength validation 4`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#454545",
        "marginBottom": 8,
      }
    }
  >
    Label
  </Text>
  <TextInput
    label="Label"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "#DDDDDD",
          "borderRadius": 3,
          "borderWidth": 1,
          "color": "#9B9B9B",
          "fontSize": 14,
          "height": 40,
          "padding": 10,
          "paddingBottom": 10,
        },
        undefined,
        Object {
          "borderColor": "red",
        },
      ]
    }
    underlineColorAndroid="transparent"
    validation={
      Object {
        "maxLength": 2,
      }
    }
    value="abC"
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "red",
        "margin": 4,
        "marginBottom": 0,
      }
    }
  >
    Label maksimum undefined karakter
  </Text>
</View>
`;

exports[`AmfInputText with minLength validation 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#454545",
        "marginBottom": 8,
      }
    }
  >
    Label
  </Text>
  <TextInput
    label="Label"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "#DDDDDD",
          "borderRadius": 3,
          "borderWidth": 1,
          "color": "#9B9B9B",
          "fontSize": 14,
          "height": 40,
          "padding": 10,
          "paddingBottom": 10,
        },
        undefined,
        Object {},
      ]
    }
    underlineColorAndroid="transparent"
    validation={
      Object {
        "minLength": 2,
      }
    }
    value={null}
  />
</View>
`;

exports[`AmfInputText with minLength validation 2`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#454545",
        "marginBottom": 8,
      }
    }
  >
    Label
  </Text>
  <TextInput
    label="Label"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "#DDDDDD",
          "borderRadius": 3,
          "borderWidth": 1,
          "color": "#9B9B9B",
          "fontSize": 14,
          "height": 40,
          "padding": 10,
          "paddingBottom": 10,
        },
        undefined,
        Object {
          "borderColor": "red",
        },
      ]
    }
    underlineColorAndroid="transparent"
    validation={
      Object {
        "minLength": 2,
      }
    }
    value=""
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "red",
        "margin": 4,
        "marginBottom": 0,
      }
    }
  >
    Label minimal 2 karakter
  </Text>
</View>
`;

exports[`AmfInputText with minLength validation 3`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#454545",
        "marginBottom": 8,
      }
    }
  >
    Label
  </Text>
  <TextInput
    label="Label"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "#DDDDDD",
          "borderRadius": 3,
          "borderWidth": 1,
          "color": "#9B9B9B",
          "fontSize": 14,
          "height": 40,
          "padding": 10,
          "paddingBottom": 10,
        },
        undefined,
        Object {},
      ]
    }
    underlineColorAndroid="transparent"
    validation={
      Object {
        "minLength": 2,
      }
    }
    value="ab"
  />
</View>
`;

exports[`AmfInputText with regex validation 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#454545",
        "marginBottom": 8,
      }
    }
  >
    Label
  </Text>
  <TextInput
    label="Label"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "#DDDDDD",
          "borderRadius": 3,
          "borderWidth": 1,
          "color": "#9B9B9B",
          "fontSize": 14,
          "height": 40,
          "padding": 10,
          "paddingBottom": 10,
        },
        undefined,
        Object {},
      ]
    }
    underlineColorAndroid="transparent"
    validation={
      Object {
        "regex": /e/,
      }
    }
    value={null}
  />
</View>
`;

exports[`AmfInputText with regex validation 2`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#454545",
        "marginBottom": 8,
      }
    }
  >
    Label
  </Text>
  <TextInput
    label="Label"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "#DDDDDD",
          "borderRadius": 3,
          "borderWidth": 1,
          "color": "#9B9B9B",
          "fontSize": 14,
          "height": 40,
          "padding": 10,
          "paddingBottom": 10,
        },
        undefined,
        Object {},
      ]
    }
    underlineColorAndroid="transparent"
    validation={
      Object {
        "regex": /e/,
      }
    }
    value=""
  />
</View>
`;

exports[`AmfInputText with regex validation 3`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#454545",
        "marginBottom": 8,
      }
    }
  >
    Label
  </Text>
  <TextInput
    label="Label"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "#DDDDDD",
          "borderRadius": 3,
          "borderWidth": 1,
          "color": "#9B9B9B",
          "fontSize": 14,
          "height": 40,
          "padding": 10,
          "paddingBottom": 10,
        },
        undefined,
        Object {},
      ]
    }
    underlineColorAndroid="transparent"
    validation={
      Object {
        "regex": /e/,
      }
    }
    value="test"
  />
</View>
`;

exports[`AmfInputText with regex validation 4`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#454545",
        "marginBottom": 8,
      }
    }
  >
    Label
  </Text>
  <TextInput
    label="Label"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "#DDDDDD",
          "borderRadius": 3,
          "borderWidth": 1,
          "color": "#9B9B9B",
          "fontSize": 14,
          "height": 40,
          "padding": 10,
          "paddingBottom": 10,
        },
        undefined,
        Object {
          "borderColor": "red",
        },
      ]
    }
    underlineColorAndroid="transparent"
    validation={
      Object {
        "regex": /e/,
      }
    }
    value="ups"
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "red",
        "margin": 4,
        "marginBottom": 0,
      }
    }
  >
    Format tidak sesuai
  </Text>
</View>
`;

exports[`AmfInputText with required validation 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#454545",
        "marginBottom": 8,
      }
    }
  >
    Label
  </Text>
  <TextInput
    label="Label"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "#DDDDDD",
          "borderRadius": 3,
          "borderWidth": 1,
          "color": "#9B9B9B",
          "fontSize": 14,
          "height": 40,
          "padding": 10,
          "paddingBottom": 10,
        },
        undefined,
        Object {},
      ]
    }
    underlineColorAndroid="transparent"
    validation={
      Object {
        "required": true,
      }
    }
    value={null}
  />
</View>
`;

exports[`AmfInputText with required validation 2`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#454545",
        "marginBottom": 8,
      }
    }
  >
    Label
  </Text>
  <TextInput
    label="Label"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "#DDDDDD",
          "borderRadius": 3,
          "borderWidth": 1,
          "color": "#9B9B9B",
          "fontSize": 14,
          "height": 40,
          "padding": 10,
          "paddingBottom": 10,
        },
        undefined,
        Object {
          "borderColor": "red",
        },
      ]
    }
    underlineColorAndroid="transparent"
    validation={
      Object {
        "required": true,
      }
    }
    value=""
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "red",
        "margin": 4,
        "marginBottom": 0,
      }
    }
  >
    Label tidak boleh kosong
  </Text>
</View>
`;

exports[`AmfInputText with required validation 3`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#454545",
        "marginBottom": 8,
      }
    }
  >
    Label
  </Text>
  <TextInput
    label="Label"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "#DDDDDD",
          "borderRadius": 3,
          "borderWidth": 1,
          "color": "#9B9B9B",
          "fontSize": 14,
          "height": 40,
          "padding": 10,
          "paddingBottom": 10,
        },
        undefined,
        Object {},
      ]
    }
    underlineColorAndroid="transparent"
    validation={
      Object {
        "required": true,
      }
    }
    value="ab"
  />
</View>
`;

exports[`AmfInputText without validation 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#454545",
        "marginBottom": 8,
      }
    }
  >
    Label
  </Text>
  <TextInput
    label="Label"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "#DDDDDD",
          "borderRadius": 3,
          "borderWidth": 1,
          "color": "#9B9B9B",
          "fontSize": 14,
          "height": 40,
          "padding": 10,
          "paddingBottom": 10,
        },
        undefined,
        Object {},
      ]
    }
    underlineColorAndroid="transparent"
    value={null}
  />
</View>
`;
